hypothesis-python/src/hypothesis/statistics.py

```diff
@@ -85,7 +87,7 @@ def __init__(self, engine):
         total_runtime = math.fsum(engine.all_runtimes)
         total_drawtime = math.fsum(engine.all_drawtimes)
 
-        if total_drawtime == 0.0:
+        if total_drawtime <= 0.0 or total_runtime <= 0.0:
```

Comment:
I assume this is floating point being weird – why would total time be less than 0?

Reply:
Alas, no - what I've actually observed failing is that total_runtime may be *equal* to zero, due to coarse VM clocks (thanks to general shared-VM issues, but also things like Spectre mitigations).  The defence against negative times is a more speculative defence that could only ever be useful because monotonic clocks are a Python3-only thing and so observed time might jump backwards :roll_eyes:

Reply:
Mmm.

Okay. Could we get a comment to that effect please, because this is decidedly non-obvious.

Reply:
Will do.